lib/SIL/SILDeclRef.cpp

```diff
@@ -381,6 +381,15 @@ SILLinkage SILDeclRef::getLinkage(ForDefinition_t forDefinition) const {
   // FIXME: @objc declarations should be too, but we currently have no way
   // of marking them "used" other than making them external. 
   ValueDecl *d = getDecl();
+
+  // stdlib_binary_only functions should have a public SIL linkage.
+  auto Attrs = d->getAttrs();
+  for (auto *A : Attrs.getAttributes<SemanticsAttr>()) {
+    if (A->Value == "stdlib_binary_only") {
+      return maybeAddExternal(SILLinkage::Public);
```

Comment:
Are they not declared as public in the source? Why not?

Reply:
We have a number of `@_semantics("stdlib_binary_only")` uses in the stdlib, which are not public. It seems like this attribute was considered to be totally orthogonal to the access level. It is kind of similar to the effect of using `@_versioned` in the sense that the object file would contain it as a public symbol.